bundler/lib/bundler/cli.rb

```diff
@@ -802,16 +802,7 @@ def self.deprecated_ext_value?(arguments)
     # should be called first, before you instantiate anything like an
     # `Installer` that'll keep a reference to the old one instead.
     def auto_install
-      return unless Bundler.settings[:auto_install]
```

Comment:
The description of this method still mentions `Bundler.settings[:auto_install]`, so I think it makes sense to keep this guard clause and remove it from `Bundler.auto_install`.